src/main/java/org/kiwiproject/dropwizard/util/jackson/StandardJacksonConfigurations.java

```diff
@@ -0,0 +1,93 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import io.dropwizard.setup.Environment;
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import org.kiwiproject.dropwizard.util.config.JacksonConfig;
+import org.kiwiproject.dropwizard.util.health.UnknownPropertiesHealthCheck;
+import org.kiwiproject.json.JaxbElementSerializer;
+import org.kiwiproject.json.JsonHelper;
+import org.kiwiproject.json.LoggingDeserializationProblemHandler;
+
+import javax.xml.bind.JAXBElement;
+
+/**
+ * Set of opinionated "standard" utilities to configure Jackson.
+ */
+@UtilityClass
+@Slf4j
+public class StandardJacksonConfigurations {
+
+    /**
+     * Registers all of the "standard" Jackson configurations.
+     *
+     * @param config        a config model containing various options for configuring Jackson
+     * @param environment   the Dropwizard environment
+     * @return The configured {@link ObjectMapper}
+     * @see StandardJacksonConfigurations#registerJacksonDeserializationProblemHandler(JacksonConfig, ObjectMapper, HealthCheckRegistry)
+     * @see StandardJacksonConfigurations#registerJacksonTimestampSerialization(JacksonConfig, ObjectMapper)
+     * @see StandardJacksonConfigurations#registerJaxbSerializer(JacksonConfig, ObjectMapper)
+     */
+    public static ObjectMapper registerAllStandardJacksonConfigurations(JacksonConfig config, Environment environment) {
+        var mapper = environment.getObjectMapper();
+
+        registerJacksonDeserializationProblemHandler(config, mapper, environment.healthChecks());
+        registerJacksonTimestampSerialization(config, mapper);
+        registerJaxbSerializer(config, mapper);
+
+        return mapper;
+    }
+
+    /**
+     * Optionally configures a {@link LoggingDeserializationProblemHandler} to log Jackson deserialization issues. This
+     * method will also optionally register a Health Check to mark the service unhealthy if there are deserialization
+     * issues.
+     *
+     * @param config        a config model containing various options for configuring Jackson
+     * @param mapper        the {@link ObjectMapper} to attach the handler to
+     * @param healthChecks  the {@link HealthCheckRegistry} to register the health check on
+     */
+    public static void registerJacksonDeserializationProblemHandler(JacksonConfig config,
+                                                                     ObjectMapper mapper,
+                                                                     HealthCheckRegistry healthChecks) {
+
+        if (config.isIgnoreButWarnForUnknownJsonProperties()) {
+            var handler = new LoggingDeserializationProblemHandler();
+            mapper.addHandler(handler);
+
+            if (config.isRegisterHealthCheckForUnknownJsonProperties()) {
+                healthChecks.register("Message Deserialization", new UnknownPropertiesHealthCheck(handler));
```

Comment:
"Message Deserialization" is pretty generic. Should the name be "Unknown JSON Properties" or similar?

Reply:
Sure